src/app/frontend/replicasetdetail/replicasetdetail_controller.js

```diff
@@ -55,4 +73,47 @@ export default class ReplicaSetDetailController {
     angular.forEach(object, function(value, key) { result += `,${key}=${value}`; });
     return result.substring(1);
   }
+
+  /**
+   * Handles replica set delete dialog.
+   * @export
+   */
+  handleDeleteReplicaSetDialog() {
+    showDeleteReplicaSetDialog(this.mdDialog_, this.deleteReplicaSet_.bind(this));
+  }
+
+  /**
+   * Callbacks used after clicking dialog confirmation button in order to delete replica set
+   * or log unsuccessful operation error.
+   */
+
+  /**
+   * Deletes replica set based on current replica set namespace and name.
+   * @private
+   */
+  deleteReplicaSet_() {
+    /** @type {!angular.Resource<!backendApi.ReplicaSetDetail>} */
+    let resource = this.resource_(
+        `/api/replicasets/${this.replicaSetDetail.namespace}/${this.replicaSetDetail.name}`);
```

Comment:
Hmm... We're creating this resource object in two places. Do you have any ideas on how to have it only in one place? Or the REST string at least.